### PR TITLE
Redesigining Loklak Search Settings

### LINKED ISSUE #924: Redesigining Loklak Search Settings
**I'm submitting a ...**
- [x] feature request

**Current behavior:**

Outdated UI which is quite lengthy and breaks the user experience. Instead, we can use responsive bootstrap cards.

### PR DESCRIPTION
**Pull request by championpaddler**
**Changes proposed in this pull request**
- Redesigning the settings component and enhancing the UI experience using bootstrap cards. 

Preview https://loklaktest.herokuapp.com/settings
**Screenshots (if appropriate)** 
screenshot from 2018-11-01 23-10-38
**Link to live demo: http://pr-926-fossasia-loklaksearch.surge.sh  

**Closes #924**

### CODE REVIEW

**Comment by championpaddler** [comment]
@simsausaurabh Please review.

**Comment by championpaddler** [comment]
@simsausaurabh @praveenojha33 @daminisatya @gabru-md @ParthS007 Please review.

File: src/app/settings/settings.component.html [hunk 1]
```diff
@@ -1,17 +1,27 @@
-<div class="outer-wrapper">
-	<div class="wrapper">
-    <!-- Source Url Config -->
-    <h1><u>URL Configuration</u></h1>
-    <div class="url-container" *ngFor="let key of urlObject">
-      <h2>{{key[0]}}</h2>
+<link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css" integrity="sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO" crossorigin="anonymous">
```
Comment (simsausaurabh) [inline src/app/settings/settings.component.html]: Isn't bootstrap already included in root html file?

File: src/app/settings/settings.component.html [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+
+<div class="container-fluid" id="main">
+  <!-- Source Url Config -->
+  <h1><u>URL Configuration</u></h1>
+  <div class="row">
+    <div class="url-container col-lg-3 col-md-6  card" *ngFor="let key of urlObject">
+      <h2>{{key[0]|uppercase}}</h2>
```
Comment (simsausaurabh) [inline src/app/settings/settings.component.html]: Keep proper spacing

File: src/app/settings/settings.component.html [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
     <h1><u>News Configuration</u></h1>
+
     <div class="url-container" *ngFor="let key of newsConfigOrgs">
       <h4 class="news">{{key}}</h4>
     </div>
-	</div>
-</div>
+  </div>
+
+</div>
```
Comment (simsausaurabh) [inline src/app/settings/settings.component.html:24]: Add EOF.

File: src/app/settings/settings.component.scss [hunk 4]
```diff
@@ -46,3 +46,20 @@ div.outer-wrapper {
...
+#main>.card:hover {
+	cursor: pointer;
+	background-color: rgb(252, 252, 122);
+    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
+}
+#main { 	
+	width:80%;
+	text-align: center;
+	margin:auto;
+}
```
Comment (simsausaurabh) [inline src/app/settings/settings.component.scss:38]: Add EOF.

File: src/app/settings/settings.component.scss [hunk 5]
```diff
@@ -46,3 +46,20 @@ div.outer-wrapper {
 		}
 	}
 }
+.card {
+	margin:30px;
+	font-size: 20px;
+	box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
+    transition: 0.3s;
```
Comment (simsausaurabh) [inline src/app/settings/settings.component.scss]: Please follow proper indentation

File: src/app/settings/settings.component.scss [hunk 6]
```diff
@@ -46,3 +46,20 @@ div.outer-wrapper {
...
+	cursor: pointer;
+}
+#main>.card:hover {
+	cursor: pointer;
+	background-color: rgb(252, 252, 122);
+    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
```
Comment (simsausaurabh) [inline src/app/settings/settings.component.scss]: Please fix it too.

**Review by simsausaurabh — CHANGES_REQUESTED** [review]
Please fix inlines.

**Comment by championpaddler** [comment]
@simsausaurabh

**Review by simsausaurabh — CHANGES_REQUESTED** [review]
Please do not mark the requested changes as resolved when they are not already resolved.

**Comment by championpaddler** [comment]
@simsausaurabh  Can you please review this one and highlight the required changes??

**Comment by championpaddler** [comment]
@simsausaurabh  @praveenojha33  Please review.

**Comment by simsausaurabh** [comment]
@sk9331657 Could you please provide any other deployment link apart from surge ?

**Comment by championpaddler** [comment]
@simsausaurabh  Sure

**Comment by championpaddler** [comment]
@simsausaurabh  http://fervent-bohr-61e734.netlify.com/

**Comment by championpaddler** [comment]
@simsausaurabh  Seems like it needs to be hosted on heroku for testing since routing is done on frontend.

**Comment by simsausaurabh** [comment]
@sk9331657 Could you set it up once and for all on Heroku to even test your further PRs?

**Comment by simsausaurabh** [comment]
@sk9331657 Please follow: https://medium.com/@hellotunmbi/how-to-deploy-angular-application-to-heroku-1d56e09c5147 for deploying (Angular project) on Heroku.

**Comment by championpaddler** [comment]
> @sk9331657 Please follow: https://medium.com/@hellotunmbi/how-to-deploy-angular-application-to-heroku-1d56e09c5147 for deploying (Angular project) on Heroku.

@simsausaurabh  I will deploy the pr's.

**Comment by championpaddler** [comment]
@simsausaurabh  Preview https://loklaktest.herokuapp.com/settings

**Comment by championpaddler** [comment]
@simsausaurabh  Please review.

**Comment by championpaddler** [comment]
@praveenojha33  Review.

**Comment by simsausaurabh** [comment]
@sk9331657 Please make active links: e.g. ```root```, etc inactive. And show the other active URLs like: ```fossasia.org``` instead of ```https://fossasia.org```, it should work as usual but it should not display other non useful matter e.g. https. Make background a little bit grey (not primarily, nor purely white) something like: 

screenshot from 2018-11-11 13-18-40

**Comment by championpaddler** [comment]
@simsausaurabh  Can you please review https://loklaktest.herokuapp.com/settings ??

**Comment by simsausaurabh** [comment]
@sk9331657 Yes, it has improved a lot, good work. Can you shorten the URLs too as mentioned in above comment by me. News Configuration card on bottom looks a bit odd, comparing to others.

**Comment by championpaddler** [comment]
@simsausaurabh @praveenojha33 Please review now.

**Review by praveenojha33 — CHANGES_REQUESTED** [review]
@sk9331657 Please fix travis.

**Comment by championpaddler** [comment]
@simsausaurabh  Please review now.

**Comment by championpaddler** [comment]
@praveen @simsausaurabh  please review and merge this now.

**Comment by championpaddler** [comment]
@simsausaurabh  @praveenojha33

**Comment by championpaddler** [comment]
@mariobehling  Can you please review this pr??

**Comment by praveenojha33** [comment]
@sk9331657 Please update the corresponding issue. Please add more information to the issue so that everyone can easily understand the issue and review the corresponding PR. :+1:

**Comment by praveenojha33** [comment]
Please also update the surge link to heroku link in PR description.

**Comment by praveenojha33** [comment]
Rest other things looks good to me. Please update issue and PR description so that it will be easy for everyone else to review your PR.

**Comment by championpaddler** [comment]
> Rest other things looks good to me. Please update issue and PR description so that it will be easy for everyone else to review your PR.

Done @praveenojha33  . Please review.

**Comment by championpaddler** [comment]
@simsausaurabh  Please review.

**Comment by championpaddler** [comment]
@fossasia/loklak-devs  Please review.

**Review by simsausaurabh — APPROVED** [review]
For start, it looks good to me.
@mariobehling Please review.

**Comment by simsausaurabh** [comment]
@sk9331657 Please also update the required details in issue description.

**Comment by championpaddler** [comment]
@mariobehling  @simsausaurabh Please review.

**Review by praveenojha33 — APPROVED** [review]
LGTM :+1:

**Comment by championpaddler** [comment]
@simsausaurabh  Reviews??

**Comment by championpaddler** [comment]
@mariobehling  Please review and merge this.

**Comment by simsausaurabh** [comment]
@sk9331657 It looks good for start. Please update the issue description which you are currently solving.

@mariobehling Please review.

**Comment by championpaddler** [comment]
@simsausaurabh  i have updated the issue description.

**Comment by simsausaurabh** [comment]
Your deployment isn't active anymore?

**Comment by championpaddler** [comment]
@simsausaurabh  DO I redeploy as heroku has limit of 5 project so I deployed other project??

**Comment by simsausaurabh** [comment]
I have requested review from @mariobehling. Please also mention the PR link on Gitter. Yes, please redeploy.

**Comment by championpaddler** [comment]
@mariobehling  Please review. Here is deployment link http://agorawebapp.herokuapp.com/

**Comment by championpaddler** [comment]
@mariobehling  Can you please review this?? This is pending since long time.

**Comment by championpaddler** [comment]
@mariobehling  Please review this also .

**Comment by simsausaurabh** [comment]
@sk9331657 Lets keep it on hold for a while. I will discuss it with co-developers and then we will decide how to proceed with this. Feel free to discuss about it on Gitter.

**Pull request closed without merging**